### ISSUE TITLE
Add a 50 % FPS update to reflection update for ' Always ' setting . .

### ISSUE THREAD

**Issue by charlesmlamb** [godotengine/godot-proposals#4739]
### Describe the project you are working on

A 2.5D game, atm . . 

### Describe the problem or limitation you are having in your project

I saw video about Reflectionprobe, super - cool . . However, FPS went quite low, in always mode, looked like down to 20 - 30 % . . Even, for a small room, or so . .

Perhaps, was thinking, could one update reflections only every other frame, to make it faster, or so . . .

I have no idea if this might work, or, could one lower resolution of the reflection map, with some setting, or so . .

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Not sure, maybe a weird idea, just wondering if there's a simple way, to get the FPS to only 80 % slower, for ' always ' update content . . .

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Well, it was just an idea, not sure . . .

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I have got no idea . . In some video - games, some animations ran at 15 FPS, and, the core game was 30 FPS, old RTS, for instance . . The result was still nice, imo a classic look, was thinking, that might work for reflection probes . .

### Is there a reason why this should be core and not an add-on in the asset library?

Well, not sure, was hoping to get a ' fast ' mode, for perhaps updating in a different way, etc . . .

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot/pull/55929.

I think we should do what clayjohn suggested and separate the current Update Mode property into 3 properties:

- **Update Mode**
  - Once
  - When Moved
  - Always
- **Update Speed**
  - Slow (1 Face per Frame)
  - Medium (2 Faces per Frame)
  - Fast (3 Faces per Frame)
  - Instant (6 Faces per Frame)
- **Filter Mode**
  - High-Quality
  - Real-Time

**Comment by charlesmlamb** [comment]
This is, really good . . 10 / 10 . . .

**Comment by jtnicholl** [comment]
Maybe we could combine these ideas and have a "very slow" update speed that updates every other frame. Half of the very slow probes in the scene would update on one frame, the other half on the next. Might be useful in a scene with lots of probes.
I don't know if that would be feasible to implement or even worth it, but it was a thought I had.

**Comment by Calinou** [comment]
> Maybe we could combine these ideas and have a "very slow" update speed that updates every other frame. Half of the very slow probes in the scene would update on one frame, the other half on the next. Might be useful in a scene with lots of probes.
I don't know if that would be feasible to implement, or if there'd be many cases it would help, but it was a thought I had.

At this point, the update speed will be so slow that reflections will look very stuttery. I'd suggest lowering the resolution of reflection probe textures instead.

**Comment by IPlayKindred** [comment]
that is true but having a stuttering reflection probe is less of an issuing than having no reflections at all in some hardware, reflection probe's performance is a bottleneck in most cases, having the ability to change the rate at which they update makes it so some other options can be checked out like higher res shadows and higher resolutions.

plus they aren't that noticeable anyway, you need to take your eyes off the road and look at the rear of the car for a second or two to realize it's running at a lower framerate.

having the option to change the update rate is very important.
